ext/standard/var.c

```diff
@@ -980,7 +980,13 @@ static void php_var_serialize_nested_data(smart_str *buf, zval *struc, HashTable
 static void php_var_serialize_class(smart_str *buf, zval *struc, zval *retval_ptr, php_serialize_data_t var_hash) /* {{{ */
 {
 	HashTable props;
-	if (php_var_serialize_get_sleep_props(&props, struc, HASH_OF(retval_ptr)) == SUCCESS) {
+
+	if (Z_TYPE_P(retval_ptr) != IS_ARRAY) {
```

Comment:
This check should be in php_var_serialize_call_sleep, replacing the HASH_OF check there.

Reply:
Right, of course.

Also, we want to do the other kind of fix (correctly) in branches below master, right ?